source/tutorials/ex_corrupt_sbl_component.rst

```diff
@@ -40,8 +40,91 @@ corruptcomp (runtime corruption tool)
 This tool is useful for testing the firmware resiliency and recovery feature
 (see :ref:`firmware-resiliency-and-recovery`).
 
-This tool is part of the OS loader shell (not the UEFI shell). To get it to show up, set PcdCmdCorruptShellAppEnabled
-to TRUE in BootloaderCommonPkg.dec and then build and stich SBL.
+This tool is part of the OS loader shell (not the UEFI shell). To get it to show up, include
+``ShellCommandRegister (Shell, &ShellCommandCorruptComp);`` in the ``LoadShellCommands`` function of ShellCmds.c.
+Also, include ``FspsConfig->PchWriteProtectionEnable[PrIndex] = FALSE;`` for each existing ``PrIndex`` in the
+``UpdateFspConfig`` function of the platform's FspsUpdUpdateLib.c.
+
+For example, in C:\\SblPlatform\\SblOpen\\BootloaderCommonPkg\\Library\\ShellLib\\ShellCmds.c, the following addition
+should be made:
+
+    .. code-block:: C
+
+        // Basic Shell commands
+        ShellCommandRegister (Shell, &ShellCommandExit);
+        ShellCommandRegister (Shell, &ShellCommandHelp);
+        ShellCommandRegister (Shell, &ShellCommandMm);
+        ShellCommandRegister (Shell, &ShellCommandCpuid);
+        ShellCommandRegister (Shell, &ShellCommandMsr);
+        ShellCommandRegister (Shell, &ShellCommandMtrr);
+        ShellCommandRegister (Shell, &ShellCommandUcode);
+        ShellCommandRegister (Shell, &ShellCommandCls);
+
+        if (!FeaturePcdGet (PcdMiniShellEnabled)) {
+            // More Shell commands
+            ShellCommandRegister (Shell, &ShellCommandPci);
+            ShellCommandRegister (Shell, &ShellCommandHob);
+            ShellCommandRegister (Shell, &ShellCommandMmap);
+            ShellCommandRegister (Shell, &ShellCommandPerf);
+            ShellCommandRegister (Shell, &ShellCommandBoot);
+            ShellCommandRegister (Shell, &ShellCommandMmcDll);
+            ShellCommandRegister (Shell, &ShellCommandCdata);
+            ShellCommandRegister (Shell, &ShellCommandDmesg);
+            ShellCommandRegister (Shell, &ShellCommandReset);
+            ShellCommandRegister (Shell, &ShellCommandFs);
+            ShellCommandRegister (Shell, &ShellCommandUsbDev);
+
+            // Load Platform specific shell commands
+            ShellExtensionCmds = GetShellExtensionCmds ();
+            for (Iter = ShellExtensionCmds; *Iter != NULL; Iter++) {
+            ShellCommandRegister (Shell, *Iter);
+            }
+        }
+
+        ShellCommandRegister (Shell, &ShellCommandCorruptComp); // Added
+
+And in SblOpen\\Platform\\AlderlakeBoardPkg\\Library\\FspsUpdUpdateLib\\FspsUpdUpdateLib.c, the following
+changes should be made:

```

Comment:
Add the purpose  why below change is required (disable flash protection so that we could corrupt flash from shell).
Maybe just list lines that need be changed, or highlighted them.